weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -527,31 +527,43 @@ def refs_read_batch(self, req: tsi.RefsReadBatchReq) -> tsi.RefsReadBatchRes:
 
         root_val_cache: typing.Dict[str, typing.Any] = {}
 
-        def get_object_ref_root_val(r: refs_internal.InternalObjectRef) -> typing.Any:
-            cache_key = f"{r.project_id}/{r.name}/{r.version}"
-            if cache_key in root_val_cache:
-                val = root_val_cache[cache_key]
-            else:
-                if r.version == "latest":
-                    raise NotFoundError("Reading refs with `latest` is not supported")
-                else:
-                    conds = [
-                        "object_id = {object_id: String}",
-                        "digest = {version: String}",
-                    ]
-                    parameters = {
-                        "object_id": r.name,
-                        "version": r.version,
-                    }
+        def make_ref_cache_key(ref: refs_internal.InternalObjectRef) -> str:
+            return f"{ref.project_id}/{ref.name}/{ref.version}"
+
+        def make_obj_cache_key(obj: SelectableCHObjSchema) -> str:
+            return f"{obj.project_id}/{obj.object_id}/{obj.digest}"
```

Comment:
This reminds me that we need to be more restrictive about the names we let people use in publishing, e.g. we should disallow ':' and '/'.